open-apparel-registry/open-apparel-registry
# Get sector from CSV upload or API request

## Overview

This PR adds a new required field, `sector`, to `FacilityListItem`, and supports setting it via either CSV upload or the API

Connects #1830 

## Demo

Not a very exciting demo because Sector doesn't show up on the front-end yet much.

It does show up in some error messages if you omit `sector` from your CSV:
![image](https://user-images.githubusercontent.com/4432106/170566965-10e9f15a-5a4d-463c-9994-0d2bf5f40617.png)

Or if it's missing from a row in the CSV:
![image](https://user-images.githubusercontent.com/4432106/170567108-9324b8f6-0cf2-4a3c-b5a8-1e131bad881e.png)


# Notes

Templates with the added sector column can be found here: [OAR_Contributor_Template_CSV_Additional_Data_Points.zip](https://github.com/open-apparel-registry/open-apparel-registry/files/8781742/OAR_Contributor_Template_CSV_Additional_Data_Points.zip)


## Testing Instructions

* `scripts/resetdb`
* `scripts/server`
* Use the Contribute page to upload a CSV. Without a `sector` field you should receive an error. Add a sector column to your CSV, fill it with a value (e.g. `Apparel`) and upload it again - it should succeed
* The `makefixtures` and `geocode_csvs` management commands work correctly, but now including a `sector` column in their input/output

## Checklist
